src/ape/managers/compilers.py

```diff
@@ -249,11 +262,12 @@ def enrich_error(self, err: ContractLogicError) -> ContractLogicError:
             return err
 
         ext = Path(contract.source_id).suffix
-        if ext not in self.registered_compilers:
+        if ext not in self.supported_extensions:
             # Compiler not found.
             return err
 
-        compiler = self.registered_compilers[ext]
+        compiler = self.get_compiler(ext)
+        assert compiler is not None
```

Comment:
i really am against getting random assertion errors in production code. if this is for mypy, add a note saying that or handle this situation differently. not to mention asserts get stripped out of optimized python.

this goes for all the asserts being requested to merge.